src/WingetCreateCLI/Commands/NewCommand.cs

```diff
@@ -115,7 +121,56 @@ public override async Task<bool> Execute()
                         return false;
                     }
 
-                    installerUpdateList.Add(new InstallerMetadata { InstallerUrl = installerUrl, PackageFile = packageFile });
+                    if (packageFile.IsZipFile())
+                    {
+                        string extractDirectory = Path.Combine(PackageParser.InstallerDownloadPath, Path.GetFileNameWithoutExtension(packageFile));
+
+                        try
+                        {
+                            ZipFile.ExtractToDirectory(packageFile, extractDirectory, true);
+                        }
+                        catch (InvalidDataException invalidDataException)
+                        {
+                            Logger.ErrorLocalized(nameof(Resources.InvalidZipFile_ErrorMessage), invalidDataException);
+                            return false;
+                        }
+
+                        List<string> extractedFiles = Directory.EnumerateFiles(extractDirectory, "*.*", SearchOption.AllDirectories)
+                            .Select(filePath => filePath = Path.GetRelativePath(extractDirectory, filePath))
+                            .Where(filePath => SupportedInstallerTypeExtensions.Contains(Path.GetExtension(filePath)))

```

Comment:
>SupportedInstallerTypeExtensions [](http://example.com/codeflow?start=47&length=32)

Do we require portables to end with exe only? This seems too strict for portables inside zip

Reply:
by design